.github/actions/cmx-versions/index.js

```diff
@@ -122,21 +124,43 @@ function getLatestVersion(distribution) {
     return latestVersion;
 }
 
-function getLatestMinorVersions(distribution) {
+function getLastMajorMinorVersions(distribution, numOfLatestMajorMinorVersions) {
+    var latestMajorMinorVersions = [];
+    distribution.versions.forEach((version) => {
+        const parsed = semverCoerce(version);
+        latestMajorMinorVersions.push(`${semverMajor(parsed)}.${semverMinor(parsed)}`);
+    });
+    latestMajorMinorVersions = sortVersions(latestMajorMinorVersions);
+    return latestMajorMinorVersions.slice(0, numOfLatestMajorMinorVersions);
+}
+
+function sortVersions(latestMajorMinorVersions) {
+    latestMajorMinorVersions = [...new Set(latestMajorMinorVersions)];
+    latestMajorMinorVersions = latestMajorMinorVersions.map(version => version + ".0");
+    latestMajorMinorVersions = semverRSort(latestMajorMinorVersions);
+    latestMajorMinorVersions = latestMajorMinorVersions.map(version => version.replace('.0', ''));
+    return latestMajorMinorVersions;
+}
+
+function getLatestMinorVersions(distribution, majorMinorVersionFilter) {
     const latestMinorVersions = {};
     distribution.versions.forEach((version) => {
         const parsed = semverCoerce(version);
+        // Check if majorVersions is null, undefined, empty, or includes the current version's major
         const majorMinor = `${semverMajor(parsed)}.${semverMinor(parsed)}`;
-        if (latestMinorVersions[majorMinor] === undefined) {
-            latestMinorVersions[majorMinor] = version;
-        } else {
-            const currentVersion = latestMinorVersions[majorMinor];
-            if (semverGt(parsed, semverCoerce(currentVersion))) {
+        if(majorMinorVersionFilter === null || majorMinorVersionFilter === undefined || majorMinorVersionFilter.size === 0 || majorMinorVersionFilter.includes(majorMinor)) {
```

Comment:
i think this can just be (tested locally and passes tests):
```suggestion
        if(!majorMinorVersionFilter?.length || majorMinorVersionFilter.includes(majorMinor)) {
```